core/templates/pages/admin-page/features-tab/admin-features-tab.component.spec.ts

```diff
@@ -0,0 +1,374 @@
+import { AdminTaskManagerService } from
+  '../services/admin-task-manager.service';
+import { PlatformFeatureAdminBackendApiService } from
+  'domain/platform_feature/platform-feature-admin-backend-api.service';
+import { PlatformParameterObjectFactory, FeatureStage, PlatformParameter } from
+  'domain/platform_feature/platform-parameter-object.factory';
+import { PlatformParameterFilterType, ServerMode } from
+  'domain/platform_feature/platform-parameter-filter-object.factory';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+
+describe('Admin page feature tab', function() {
+  let component: AdminFeaturesTabComponent;
+  let fixture: ComponentFixture<AdminFeaturesTabComponent>;
+
+  let paramFactory: PlatformParameterObjectFactory;
+  let adminDataService: AdminDataService;
+  let featureApiService: PlatformFeatureAdminBackendApiService;
+  let adminTaskManagerService: AdminTaskManagerService;
+  let windowRef: WindowRef;
+
+  let updateApiSpy: jasmine.Spy;
+
+  let mockConfirmResult: (val: boolean) => void;
+  let mockPromptResult: (msg: string) => void;
+
+  beforeEach(async(() => {
+    TestBed
+      .configureTestingModule({
+        imports: [FormsModule, HttpClientTestingModule],
+        declarations: [AdminFeaturesTabComponent],
+      })
+      .compileComponents();
+
+    paramFactory = TestBed.get(PlatformParameterObjectFactory);
+    adminDataService = TestBed.get(AdminDataService);
+    featureApiService = TestBed.get(PlatformFeatureAdminBackendApiService);
+    windowRef = TestBed.get(WindowRef);
+    adminTaskManagerService = TestBed.get(AdminTaskManagerService);
+
+    let confirmResult = true;
+    let promptResult = 'mock msg';
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      confirm: () => confirmResult,
+      prompt: () => promptResult
+    });
+    mockConfirmResult = val => confirmResult = val;
+    mockPromptResult = msg => promptResult = msg;
+
+    spyOn(adminDataService, 'getDataAsync').and.resolveTo(<AdminPageData>{
+      featureFlags: [
+        paramFactory.createFromBackendDict({
+          data_type: 'bool',
+          default_value: false,
+          description: 'This is a dummy feature flag.',
+          feature_stage: FeatureStage.DEV,
+          is_feature: true,
+          name: 'dummy_feature',
+          rule_schema_version: 1,
+          rules: [{
+            filters: [
+              {
+                type: PlatformParameterFilterType.ServerMode,
+                conditions: [['=', ServerMode.Dev]]
+              }
+            ],
+            value_when_matched: true,
+          }],
+        })
+      ]
+    });
+
+    updateApiSpy = spyOn(featureApiService, 'updateFeatureFlag')
+      .and.resolveTo(null);
+  }));
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(AdminFeaturesTabComponent);
+    component = fixture.componentInstance;
+
+    component.ngOnInit();
+  }));
```

Comment:
Can this be added to the above beforeEach function?

Reply:
Yes, combined the two beforeEach functions.